src/test/java/org/pesmypetcare/webservice/controller/petmanager/PetMealRestControllerTest.java

```diff
@@ -0,0 +1,133 @@
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.BDDMockito.willDoNothing;
+import static org.mockito.BDDMockito.willReturn;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ExtendWith(MockitoExtension.class)
+public class PetMealRestControllerTest {
+    private static String jsonMealEntity;
+    private static String jsonField;
+    private static MealEntity mealEntity;
+    private static List<Map<String, Object>> mealEntityList;
+    private static String owner;
+    private static String petName;
+    private static String date;
+    private static String date2;
+    private static String field;
+    private static String value;
+    private static String urlBase;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private MealService service;
+
+    @BeforeAll
+    public static void setUp() {
+        jsonMealEntity = "{\n"
+            + "  \"mealName\": \"Asparagus with milk\",\n"
+            + "  \"kcal\": 60.0\n"
+            + "}";
+        jsonField = "{\n"
+            + "  \"value\": \"Asparagus with chocolate\"\n"
+            + "} ";
+        mealEntity = new MealEntity();
+        mealEntityList = new ArrayList<>();
+        owner = "Manolo Manolon";
+        petName = "Canpeque2";
+        date = "2019-02-13T10:30:00";
+        date2 = "2021-02-13T10:30:00";
+        field = "mealName";
+        value = "Asparagus with ketchup";
+        urlBase = "/pet/meal";
+    }
+
+    @Test
+    public void createMealShouldReturnStatusOk() throws Exception {
+        willDoNothing().given(service).createMeal(anyString(), anyString(), anyString(), isA(MealEntity.class));
+        mockMvc.perform(post(urlBase + "/" + owner + "/" + petName + "/" + date)
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(jsonMealEntity))
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    public void deleteByDateShouldReturnStatusOk() throws Exception {
+        willDoNothing().given(service).deleteByDate(anyString(), anyString(), anyString());
+        mockMvc.perform(delete(urlBase + "/" + owner + "/" + petName + "/" + date)
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    public void deleteAllMealsShouldReturnStatusOk() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Throwing 'Exception' is not allowed.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5301962)